### PR TITLE
private-events-features

### PR DESCRIPTION
**Pull request by edemagbenyo**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = 'Welcome to our private events platform'
+      redirect_to @user
```
Comment (mosaaleb) [inline app/controllers/users_controller.rb:13]: - [x] Let's sign in the user automatically after user creation.

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<h2>Past events</h2>
+<ul>
+<% Event.past.each do |event| %>
```
Comment (mosaaleb) [inline app/views/events/index.html.erb]: - [x] There should be no application logic in the views -- `Event.past` and `Event.future` are business logic and should not be here.
You are already sending instance variables from `index` action in `events_controller`.
Move the bussines logic in the controller as instance variable and use them directly in here.

File: app/views/events/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<h1>Event information</h1>
+<p>Event Creator: <%= @event.user.name %></p>
+<h2>Attendees</h2>
+<ul>
+  <% @event.attendances.each do |a| %>
+    <li><%= User.find_by(id: a.user_id).name %></li>
```
Comment (mosaaleb) [inline app/views/events/show.html.erb]: - [x] Same here for business logic `User.find_by` shouldn't be in views.
Also we can simply loop through `@event.attendees`, you have already defined an association for it in `event.rb`:
`has_many :attendees, through: :attendances`

**Review by mosaaleb — CHANGES_REQUESTED** [review]
## Well done Team :clap:

### Most requirements are handled, yet we need to work on other requirements before moving on:

- [x] Creating events with no user signed in breaks the application:

fail

we need to make sure that only logged in users can create or attend an event. We can restrict the request to creating and attending events routes by using [filters](https://guides.rubyonrails.org/action_controller_overview.html#filters).

In our case this could be something like in `events_controller`:
`before_action :require_login, only: [:action_name_here_that_you_want_to_restrict]`

and `require_login` can be defined in `application_controller` as:
```ruby
def require_login
  redirect_to signin_path unless logged_in?
end
```
This way If a logged out user tried to deliberately type `events/new` to create an event rails will redirect him to user sign in page.

We need also to apply this to event attendance functionality

- [ ] I can't find a way for a user to attend an event -- If you believe that you implemented this requirements, contact me on slack. Otherwise, please let's add it.


- [ ] We haven't written any rspec and capybara tests yet:





#### Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

**Comment by Grifo89** [comment]
@mosaaleb Can you show us where are those requirements you are talking about? remember you must stick on the requirements that are explicitly on the project page, not less not more, more must be optional.

**Comment by Grifo89** [comment]
Changes suggested are done!

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Very nice work on this project, however, the following issues need to be fixed before moving on ⬇️** 

- You have to use Git Flow: Pull Requests from feature branches to the `develop` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/git_flow_requirement.png) from Pathwright).

- Pull Request should have a meaningful title and a short description ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)).


### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Comment by edemagbenyo** [comment]
Hi @Redvanisation ,
About the creation of pull request using gitflow, It was a mistake from our part from the onset. It seems not clear how we can do that now that we have implemented the changes in the develop branch. Do you have any suggestions for us to correct it?

**Pull request closed without merging**